### DISCUSSION TITLE
Bring back Select and MultiSelect functionality from v6

### DISCUSSION THREAD

**Discussion by juliuslissau** [mantinedev/mantine#5185] [Feature requests]
Would be amazing if the `Select` and `MultiSelect` components in version 7 could bring back the functionality of what they could do in version 6.

Specifically having the `data` prop being able to receive an array of objects, so you can easily separate the label and value of each option.

Also the `itemComponent` prop was great for easily creating custom designs for the options.

Just bringing back these two features would be amazing!

One of the main reasons I love Mantine is all the form features. Meaning the input fields, calendar, dropzone and so on - combined with use-form.

But building custom Combobox inputs and also manually having to make them work together with use-form is very difficult and time consuming for me.

---

EDIT: ... Hmm, it seems it's already possible to insert separate `label` and `value` for the data options. Not sure if I missed something in the docs or messed up my code in another way yesterday when I was working on this.

But anyway getting the `itemComponent` prop back would still be amazing!

**Comment by medihack** [comment]
I want to second this regarding the `itemComponent`. Don't get me wrong. `Combobox` is really great for complete customization, but I often just want to add an image to a select option. For that simple task, `itemComponent` was wonderful.

**Comment by jt561** [comment]
I'm planning to build a wrapper around these components to add the features that were lost. If anyone has any tips, please share them here. I'll post my progress as I go along
